product.go

```diff
@@ -152,3 +165,23 @@ func (s *ProductServiceOp) DeleteMetafield(productID int64, metafieldID int64) e
 	metafieldService := &MetafieldServiceOp{client: s.client, resource: productsResourceName, resourceID: productID}
 	return metafieldService.Delete(metafieldID)
 }
+
+// Publish or UnPublish  an existing product
+func (s *ProductServiceOp) PublishOrUnPublish(productID int64, isPublish bool) (*Product, error) {
+	path := fmt.Sprintf("%s/%s/%d.json", globalApiPathPrefix, productsBasePath, productID)
+	if isPublish {
+		now := time.Now()
+		wrappedData := ProductResource{Product: &Product{ID: productID, PublishedAt: &now}}
+		resource := new(ProductResource)
+		err := s.client.Put(path, wrappedData, resource)
+		return resource.Product, err
+
+	} else {
+		wrappedData := ProductPublishResource{Product: &ProductPublish{ID: productID, PublishedAt: nil}}
```

Comment:
Lines 181-183 are duplicating lines 175-177 so they can be moved outside of the if clause, but as I mentioned above it will be better to have 2 methods instead of one here